File: Python/ValidAnagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if(len(s)!=len(t)):
            return False
        sDict = {}
        tDict = {}
        for i in range(len(s)):
            if s[i] not in sDict:
                sDict[s[i]] = 1
            else:
                sDict[s[i]] += 1
            if t[i] not in tDict:
                tDict[t[i]] = 1
            else:
                tDict[t[i]] += 1
        return True if (sDict == tDict) else False
    # To determine if t is an anagram of s, we get the frequency dictionaries of both s and t, then check if they are equal
    # If equal, then it is an anagram

File: C++/ValidAnagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size() != t.size()){
            return false;
        }
        map<char, int> sCount;
        map<char, int> tCount;
        for(int i=0; i<s.size(); i++){
            if(sCount.contains(s[i])){
                sCount[s[i]]+=1;
            }
            else{
                sCount[s[i]]=1;
            }
        }
        for(int i=0; i<t.size(); i++){
            if(tCount.contains(t[i])){
                tCount[t[i]]+=1;
            }
            else{
                tCount[t[i]]=1;
            }
        }
        return (sCount == tCount);
    }
};